addons/mail_bot/static/src/owl/components/messaging_menu/messaging_menu_tests.js

```diff
@@ -0,0 +1,200 @@
+    assert.expect(2);
+
+    window.Notification.permission = 'granted';
+
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'channel_fetch_preview') {
+                return [];
+            }
+            return this._super(...arguments);
+        },
+    });
+
+    const messagingMenu = document.querySelector('.o_MessagingMenu');
+
+    assert.containsNone(messagingMenu, '.o_MessagingMenu_counter',
+        "should not display a notification counter next to the messaging menu"
+    );
+
+    messagingMenu.querySelector('.o_MessagingMenu_toggler').click();
+    await afterNextRender();
+    assert.containsNone(messagingMenu, '.o_NotificationRequest',
+        "should display no notification in the messaging menu"
+    );
+});
+
+QUnit.test('mail_bot: MessagingMenu: respond to notification prompt (denied)', async function (assert) {
+    assert.expect(3);
+
+    window.Notification.permission = 'default';
+    window.Notification.requestPermission = () => {
+        window.Notification.permission = 'denied';
+        return makeTestPromise().resolve(window.Notification.permission);
+    };
+
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'channel_fetch_preview') {
+                return [];
+            }
+            return this._super(...arguments);
+        },
+    });
+
+    const messagingMenu = document.querySelector('.o_MessagingMenu');
+
+    messagingMenu.querySelector('.o_MessagingMenu_toggler').click();
+    await afterNextRender();
+
+    messagingMenu.querySelector('.o_NotificationRequest').click();
+    await afterNextRender();
+
+    assert.containsOnce(document.body, '.toast .o_notification_content',
+        "should display a toast notification with the deny confirmation"
+    );
+    assert.containsNone(messagingMenu, '.o_MessagingMenu_counter',
+        "should not display a notification counter next to the messaging menu"
+    );
+
+    messagingMenu.querySelector('.o_MessagingMenu_toggler').click();
+    await afterNextRender();
+
+    assert.containsNone(messagingMenu, '.o_NotificationRequest',
+        "should display no notification in the messaging menu"
+    );
+});
+
+QUnit.test('notification_alert widget: display blocked notification alert', async function (assert) {
+    assert.expect(2);
+
+    window.Notification.permission = 'denied';
+
+    var form = await createView({
+        View: FormView,
+        model: 'mail.message',
+        data: this.data,
+        arch: '<form>' +
+                '<widget name="notification_alert"/>' +
+            '</form>',
+    });
+    assert.containsOnce(form, '.o_notification_alert', "Blocked notification alert should be displayed");
+
+    window.Notification.permission = 'granted';
+    await form.reload();
+    assert.containsNone(form, '.o_notification_alert', "Blocked notification alert should not be displayed");
+
+    form.destroy();
+});
```

Comment:
doesn't make sense for this test to be here.
doesn't test messaging menu component at all.